chainer/__init__.py

```diff
@@ -4,16 +4,26 @@
 import threading
 import warnings
 
+from chainer import cuda  # NOQA
+from chainer import dataset  # NOQA
+from chainer import datasets  # NOQA
 from chainer import flag
 from chainer import function
 from chainer import function_set
+from chainer import functions  # NOQA
```

Comment:
What is the motivation for adding import of functions, links, optimizers, and serializers?